cylc/flow/parsec/util.py

```diff
@@ -120,11 +120,14 @@ def listjoin(lst, none_str=''):
 
 
 def printcfg(cfg, level=0, indent=0, prefix='', none_str='',
-             handle=sys.stdout):
+             handle=None):
     """Pretty-print a parsec config item or section (nested dict).
 
     As returned by parse.config.get().
     """
+    # Note: default handle=sys.stdout, but explicitly setting this interferes
+    # with pytest capsys:
+    # https://github.com/pytest-dev/pytest/issues/1132#issuecomment-147506107
```

Comment:
I think it's just the default kwarg value that is messing with pytest, could do this:

```suggestion
    if not handle:
        handle = sys.stdout
```

Reply:
Unfortunately not, in the issue linked in the comment, it says that redefinition of `sys.stdout` is the problem

Reply:
This works for me:

```python
import sys


def whatever(handle=None):
    if not handle:
        handle = sys.stdout
    handle.write('hello world')


def test_whatever(capsys):
    whatever()
    assert capsys.readouterr().out == 'hello world'
```

Reply:
Oh, the same test fails for a different reason in that case. The capsys replacement for `sys.stdout` is not an instance of `StringIO`, but doesn't have an attribute `.mode`, which causes an exception at this point
https://github.com/cylc/cylc-flow/blob/4b5e8a98b4c9d833ffdec9d67a5c30f5509db988/cylc/flow/parsec/util.py#L142-L143

Could instead do?
```python
if hasattr(handle, 'mode') and 'b' in handle.mode:
    msg = msg.encode()
```